huggingface_pipelines/text.py

```diff
@@ -0,0 +1,355 @@
+        target_lang (str): The target language code for the texts to be decoded.
+        columns (List[str]): List of column names in the input data to process.
+        output_column_suffix (str): Suffix to append to the output column names.
+        batch_size (int): Number of items to process in each batch.
+        device (str): The device to use for computation (e.g., 'cpu' or 'cuda').
+
+
+    Example:
+
+            config = EmbeddingToTextPipelineConfig(
+            decoder_model="text_sonar_basic_encoder",
+            target_lang="eng_Latn",
+            columns=["text"],
+            output_column_suffix="embedding",
+            batch_size=32,
+            device="cuda"
+        )
+
+    """
+    decoder_model: str = "text_sonar_basic_decoder"
+    target_lang: str = "eng_Latn"
+
+    def with_overwrites(self, overwrites: EmbeddingToTextOverwrites) -> 'EmbeddingToTextPipelineConfig':
+        """
+        Create a new configuration with the specified overwrites.
+
+        Args:
+            overwrites (EmbeddingToTextOverwrites): Overwrite values for the configuration.
+
+        Returns:
+            EmbeddingToTextPipelineConfig: A new configuration instance with applied overwrites.
+        """
+        return replace(self, **overwrites)
+
+
+class HFEmbeddingToTextPipeline(Pipeline):
+    def __init__(self, config: EmbeddingToTextPipelineConfig):
+        super().__init__(config)
+        logger.info("Initializing embedding to text model...")
+        self.config = config
+        self.t2t_model = EmbeddingToTextModelPipeline(
+            decoder=self.config.decoder_model,
+            tokenizer=self.config.decoder_model,
+            device=self.config.device
+        )
+        logger.info("Model initialized.")
+
+    def process_batch(self, batch: Dict[str, Any]) -> Dict[str, List[str]]:
+        """
+        Process a batch of embeddings and convert them to text.
+        Args:
+            batch (Dict[str, Any]): Input batch containing embeddings.
+        Returns:
+            Dict[str, List[str]]: Processed batch with decoded texts.
+        """
+        for column in self.config.columns:
+            embeddings = batch[column]
+            assert all(isinstance(item, list) for item in embeddings), \
+                f"Column {column} must contain only lists of embeddings, not individual embeddings."
+            all_embeddings = np.vstack([np.array(embed)
+                                       for item in embeddings for embed in item])
+            all_decoded_texts = self.decode_embeddings(all_embeddings)
+
+            reconstructed_texts = []
+            start_idx = 0
+            for item in embeddings:
+                end_idx = start_idx + len(item)
+                reconstructed_texts.append(
+                    all_decoded_texts[start_idx:end_idx])
+                start_idx = end_idx
+
+            batch[f"{column}_{self.config.output_column_suffix}"] = reconstructed_texts
+            logger.info(
```

Comment:
Do we want to have this extensive info at every batch ? Maybe `logger.debug` instead ?